ludwig/features/bag_feature.py

```diff
@@ -94,16 +94,15 @@ def add_feature_data(
 
 @register_input_feature(BAG)
 class BagInputFeature(BagFeatureMixin, InputFeature):
-    encoder = "embed"
-    vocab = []
+    encoder = {TYPE: "embed", "vocab": []}
```

Comment:
I don't like the presence of this encoder dict in general, in part for reasons described elsewhere, but also because it mixes config with state. `vocab` is state that needs to be tracked by the feature (possibly within the encoder object). The encoder type chosen by the user is config, and should ideally live as part of the schema as well (we need to make sure changing this default doesn't break old models).

Reply:
With the full encoder defaults defined in the feature schemas, are these class-global features still needed?

Reply:
No I am going to remove most of them, the ones that seemingly need to stay for now are `metric_functions` and `default_validation_metric` since the parent class OutputFeature has them as abstract methods.